iModelCore/ECPresentation/Source/Shared/Queries/CustomFunctions.cpp

```diff
@@ -451,6 +451,13 @@ struct EvaluateECExpressionScalar : CachingScalarFunction<bmap<ECExpressionScala
             ECExpressionsCache noCache;
             if (ECExpressionsHelper(noCache).EvaluateECExpression(value, expression, *expressionContext) && value.IsPrimitive() && value.ConvertPrimitiveToString(expressionResult))
                 {
+                LabelDefinitionPtr labelDefinition = LabelDefinition::FromString(expressionResult.c_str());
+                if (labelDefinition->IsDefinitionValid())
+                    {
+                    Utf8StringCR labelRef = labelDefinition->GetDisplayValue();
+                    expressionResult = labelRef;
+                    }

```

Comment:
Why is this needed?

Reply:
GetRelatedDisplayLabel returns stringified json object, that looks like this:
"{\"DisplayValue\":\"lc1\",\"TypeName\":\"string\",\"RawValue\":\"lc1\"}"

Reply:
I think that's a problem - we may receive an ECExpression like `GetRelatedDisplayLabel("x", "y", "z") & "_ABC"` and it should concatenate label of `z` with string `_ABC`. With what we have now, `GetRelatedDisplayLabel` will return an stringified JSON object and add the `_ABC` suffix to it. This will result in an invalid JSON, so `LabelDefinition` won't parse it, so the end result will be something like `{ "DisplayValue": "", "RawValue": "", "TypeName": "string" }_ABC`.

IMO, `LabelDefinition` should be parsed in `ECInstanceMethodSymbolsProvider::GetRelatedDisplayLabel`. But first we need a test for this case.

Reply:
Moved parsing to GetRelatedDisplayLabel and added `ContentInstancesOfSpecificClassesSpecification_GetsRelatedLabelWithAppendedValueThroughCalculatedProperty` test which checks the scenario you've mentioned.